rholang/src/test/scala/coop/rchain/rholang/interpreter/accounting/RholangMethodsCostsSpec.scala

```diff
@@ -0,0 +1,827 @@
+    }
+
+  }
+
+  "interpolate" when {
+    "called on String with a Map of values to interpolate with" should {
+      "charge proportionally to the length of base String and size of the Map" in {
+        implicit val errorLog = new ErrorLog()
+        def strMapN(n: Int): Map[Par, Par] =
+          (1 to n).map(i => (GString(s"key$i"): Par, GInt(i): Par)).toMap
+
+        val data = Table(
+          ("string", "map"),
+          ("", emptyMap),
+          (stringN(100), emptyMap),
+          (stringN(100), strMapN(10))
+        )
+        forAll(data) {
+          case (string, map) =>
+            val method       = EPercentPercent(GString(string), toRholangMap(map))
+            implicit val env = Env[Par]()
+            withReducer { reducer =>
+              reducer.evalExprToPar(method).runSyncUnsafe(1.second)
+              val cost = methodCallCost(reducer)
+              assert(cost === interpolateCost(string.length, map.size))
+            }
+        }
+      }
+    }
+  }
+
+  def methodCall(method: String, target: Par, arguments: List[Par]): Expr =
+    EMethod(method, target, arguments)
+
+  def methodCallCost(reducer: ChargingReducer[Task]): Cost =
+    reducer
+      .getAvailablePhlos()
+      .map(ca => Cost(Integer.MAX_VALUE) - ca.cost - METHOD_CALL_COST)
+      .runSyncUnsafe(1.second)
+
+  def map(pairs: Seq[(Par, Par)]): Map[Par, Par] = Map(pairs: _*)
+  def emptyMap: Map[Par, Par]                    = map(Seq.empty[(Par, Par)])
+  def mapN(n: Int, value: Par): Map[Par, Par] =
+    (1 to n).map(i => (Par().withExprs(Seq(GInt(i))), value)).toMap
+  def toRholangMap(map: Map[Par, Par]): EMapBody =
+    EMapBody(ParMap(SortedParMap(map)))
+
+  def emptySet: Set[Par] = setN(0)
+  def setN(n: Int): Set[Par] =
+    (1 to n).map(i => Par().withExprs(Seq(GInt(i)))).toSet
+  def toRholangSet(set: Set[Par]): ESetBody =
+    ESetBody(ParSet(set.toSeq))
+
+  def listN(n: Int): Vector[Par] =
+    (1 to n).map(i => Par().withExprs(Vector(GInt(i)))).toVector
+  def emptyList: Vector[Par] = Vector.empty[Par]
+
+  def toRholangList(vector: Vector[Par]): EListBody =
+    EListBody(EList(vector))
+
+  def gbyteArray(n: Int): GByteArray =
+    GByteArray(ByteString.copyFrom(new Array[Byte](n)))
+
+  def stringN(n: Int): String =
+    Seq.fill(n)("a").mkString
+
+  def emptyString: String = ""
+
+  def withReducer[R](f: ChargingReducer[Task] => R)(implicit errLog: ErrorLog): R = {
+    val reducer = RholangOnlyDispatcher.create[Task, Task.Par](space)._2
+    reducer.setAvailablePhlos(Cost(Integer.MAX_VALUE)).runSyncUnsafe(1.second)
+    f(reducer)
+  }
+
+  private var dbDir: Path         = null
+  private var context: RhoContext = null
+  private var space: RhoISpace    = null
+
+  override protected def beforeAll(): Unit = {
+    import coop.rchain.rholang.interpreter.storage.implicits._
+    implicit val syncF: Sync[Id] = coop.rchain.catscontrib.effect.implicits.syncId
+    dbDir = Files.createTempDirectory("rholang-interpreter-test-")
```

Comment:
In memory representation of the LMDB? Maybe but we still have the issue of creating the `RSpace` before all and tearing down after all tests. Since we never `produce` or `consume` recreating RSpace for each test is unnecessary.

Reply:
inMemory is more suitable for unittests as you don't deal with file storage.

Reply:
OK I used the `inMemory` version.